core-web/libs/edit-content/src/lib/components/dot-edit-content-form/dot-edit-content-form.component.ts

```diff
@@ -130,24 +181,128 @@ export class DotEditContentFormComponent implements OnInit {
                     .replace(/T|\.\d{3}Z/g, (match: string) => (match === 'T' ? ' ' : '')); // To remove the T and .000Z from the date)
             }
         });
+
         this.changeValue.emit(value);
     }
 
     /**
-     * Initializes a form control for a given DotCMSContentTypeField.
+     * Processes the form value, applying specific transformations for certain field types.
+     *
+     * @private
+     * @param {Record<string, any>} value The raw form value
+     * @returns {Record<string, string>} The processed form value
+     * @memberof DotEditContentFormComponent
+     */
+    private processFormValue(
+        value: Record<string, string | string[] | Date | null | undefined>
+    ): Record<string, string> {
+        return Object.fromEntries(
+            this.$filteredFields().map(({ variable, fieldType }) => {
+                let fieldValue = value[variable];
+
+                if (
+                    Array.isArray(fieldValue) &&
+                    FLATTENED_FIELD_TYPES.includes(fieldType as FIELD_TYPES)
+                ) {
+                    fieldValue = fieldValue.join(',');
+                } else if (
+                    fieldValue instanceof Date &&
+                    CALENDAR_FIELD_TYPES.includes(fieldType as FIELD_TYPES)
+                ) {
+                    fieldValue = fieldValue
+                        .toISOString()
+                        .replace(/T|\.\d{3}Z/g, (match) => (match === 'T' ? ' ' : ''));
+                }
+
+                return [variable, fieldValue?.toString() ?? ''];
+            })
+        );
+    }
+
+    /**
+     * Initializes the form by creating form controls for each field in the content type.
+     * Skips filtered types.
      *
      * @private
-     * @param {DotCMSContentTypeField} field
-     * @return {*}
      * @memberof DotEditContentFormComponent
      */
-    private initializeFormControl(field: DotCMSContentTypeField): FormControl {
-        const validators = [];
+    private initializeForm() {
+        this.form = this.#fb.group({});
+        this.$store.contentType().fields.forEach((field) => {
```

Comment:
Add ? in this.$store.contentType()?.fields because contentType could be null.

Reply:
same as the other comment, It is unnecessary; `contentType` could never be `null`. This creates the shape of tabs that contain the `layout` property and the final `form` (`FormBuilder`).